tests/ui/needless_character_iteration.rs

```diff
@@ -0,0 +1,26 @@
+#![warn(clippy::needless_character_iteration)]
+#![allow(clippy::unnecessary_operation)]
+
+fn main() {
+    "foo".chars().all(|c| c.is_ascii());
```

Comment:
Can you add a test where there is a different method between `chars()` and `all()`? Also some test where the all call tests something else as well, like `.all(|c| c.is_ascii() && c.is_alphabetic())`

Reply:
Done!